### DISCUSSION TITLE
DockerHub authentication using encrypted password

### DISCUSSION THREAD

**Discussion by les2** [renovatebot/renovate#11865] [Request Help]
We host private images in a dockerhub registry. I'm hoping to try the following configuration, an example of which I cannot attest as being valid in the docs:

`myDockerhub.json`
```
{
  "encrypted": {
    "dockerPassword": "encrypted-value"
  },
  "hostRules": [
    {
      "hostType": "docker",
      "username": "myusername",
      "password": "${DOCKER_PASSWORD}"
    }
  ]
}
```

Which I will extend in my `renovate-config/default.json`.

Will this work?

I'm referencing documentation from https://docs.renovatebot.com/configuration-options/#hostrules
and here:
<img width="721" alt="Screen Shot 2021-09-23 at 15 18 35" src="https://user-images.githubusercontent.com/1066080/134535742-d4582b7c-6678-4f18-adc8-27791349e617.png">

**Comment by viceice** [comment]
If you are using the hosted app, you need this:
```json
{
  "hostRules": [
    {
      "hostType": "docker",
      "matchHost": "docker.io",
      "username": "myusername",
      "encrypted": {
        "password": "encrypted-value"
      }
    }
  ]
}
```

you should add `matchHost` otherwise renovate would send the auth to any docker registry

**Comment by gberche-orange** [comment]
Also supports encrypting the username for "paranoids":

```json
{
  "hostRules": [
    {
      "hostType": "docker",
      "matchHost": "docker.io",
      "encrypted": {
        "username": "encrypted-username",
        "password": "encrypted-value"
      }
    }
  ]
}
```
